RegistroCliente

RomeoCirel opened the pull request:
deberíamos ya cerrar esta rama  (si esta bien), porque ya necesitamos, y haría otra rama para los test

RomeoCirel committed:
ADD: migración crear tabla mail

RomeoCirel committed:
ADD: unique Index a la migration create tabla cliente_mail

RomeoCirel committed:
ADD Migration: add time stamps y soft delete a tabla clientes

RomeoCirel committed:
ADD Migration: add time stamps y soft delete a tabla con_clientes

RomeoCirel committed:
ADD Migrations: Update created_at & deleted_at en tabla con_clientes

RomeoCirel committed:
ADD Migrations: Update created_at & deleted_at en tabla clientes

RomeoCirel committed:
ADD migration: procedure para copiar datos de con_cliente a cliente_mail

RomeoCirel committed:
ADD migration: trigger BEFORE Update en tabla con_cliente

mantiene sincronizados los UPDATES entre las tablas con_cliente y cliente_mails, ademas de sincronizar estado y deleted_at entre el sistema nuevo y el viejo

RomeoCirel committed:
ADD migration: trigger AFTER INSERT en tabla con_cliente

Replica la inserción de un Mail en con_cliente a la tabla cliente_mail

RomeoCirel committed:
ADD Cliente: Modelo y Controller

RomeoCirel committed:
ADD Mail: Modelo y Controller

RomeoCirel committed:
Fix MailController: mensaje error on update

RomeoCirel committed:
ADD Telefono: Modelo y Controller

RomeoCirel committed:
FIX migrations: renombra el campo email a mail en la tabla cliente_mails

RomeoCirel committed:
ADD migration: add time stamps y soft delete a la tabla razones_sociales

RomeoCirel committed:
ADD migration: add time stamps y soft delete a la tabla cliente_razon

RomeoCirel committed:
ADD Migrations: Update created_at & deleted_at en tabla razones_sociales

RomeoCirel committed:
ADD migration: trigger Before Update en tabla razones_sociales

sincroniza los eliminados y rehailitados  entre el sistema viejo y el nuevo

RomeoCirel committed:
ADD Migrations: Update created_at & deleted_at en tabla cliente_razon

RomeoCirel committed:
ADD migration: trigger Before Update en tabla cliente_razon

mantiene sincronizados las eliminaciones y altas entre los 2 sistemas

RomeoCirel committed:
ADD RazonSocial: Modelo y Controlador

RomeoCirel committed:
FIX: codigos de error en MailController y TelefonoController

RomeoCirel committed:
ADD: rutas para Mail, Cliente, Telefono y RazonesSociales,

RomeoCirel committed:
FIX: renombra los modelos Telefono, mail, razonSocial y domicilio

RomeoCirel committed:
ADD: modelos Localidad y Provincia

RomeoCirel committed:
ADD: agrega las relaciones en cliente

RomeoCirel committed:
ADD: los controladores de Localidad y Provincia

P/D: con el rename de los controladores git se mareo, en realidad son nuevos y git marca como modificados

RomeoCirel committed:
ADD: controladores para mail, telefono, razones sociales y domicilio

RomeoCirel committed:
FIX: rutas para los nuevos controladores

RomeoCirel committed:
ADD  timestamps y softdelete a tabla do_cliente localidades y provincias

RomeoCirel committed:
ADD Migration: elimina index unique a telefono

RomeoCirel committed:
ADD Migration: elimina unique index mail en tabla con_cliente

RomeoCirel committed:
ADD Migration: crea trigger before Insert tabla razones_socilaes

establese el campo estado como 1 (activo) al crear una razon nueva

RomeoCirel committed:
ADD: establese la relacion entre User y Cliente

RomeoCirel committed:
ADD: bind para poder inyectar modelos eliminados en los controladores

RomeoCirel committed:
sincro notebook

RomeoCirel committed:
Merge desde develope a RegistroClientes

RomeoCirel committed:
ADD: Eloquence Mappable 

para renombrar las propiedades de los modelos

RomeoCirel committed:
ADD: Eloquence Mappable a modelos relacionados con Cliente

Cliente, ClienteMail, ClienteTelefono, ClienteDomicilio y ClienteRazonSocial.

RomeoCirel committed:
ADD: Eloquence Mappable los modelos Localidad y Provincia

RomeoCirel committed:
FIX: ClienteTelefonoController

RomeoCirel committed:
FIX: ClienteRazonSocialController

RomeoCirel committed:
FIX: ClienteMailController

RomeoCirel committed:
FIX: ProvinciaController

RomeoCirel committed:
FIX: ClienteRazonSocialRequest

cambia nombre por denominación y el CUIT es del tipo STRING

RomeoCirel committed:
FIX: ClientesController

RomeoCirel committed:
FIX: ClienteDomicilioController

RomeoCirel committed:
ADD: agrega las rutas para los controladores relacionados con CLiente

Cliente, ClienteMail, ClienteTelefono, ClienteDomicilio y ClienteRazonSocial.

RomeoCirel committed:
FIX RouteServiceProvaider: Inyeccion de ClienteDomicilio con eliminados

RomeoCirel committed:
Fix ProvinciaController index

RomeoCirel committed:
Fix: LocalidadController

RomeoCirel committed:
ADD: agrega las rutas para Provincias y Localidades

RomeoCirel committed:
Fix: agrega unos saltos de linea

MatiasOlivera committed:
style: corregir case en migraciones

MatiasOlivera commented on server/database/migrations/2019_03_20_154529_update_tabla_con_cliente_created_at.php:
```diff
@@ -0,0 +1,54 @@
+                        SET
+                            con_cliente.created_at = clientes.created_at
+                        WHERE
+                                con_cliente.id > 0
```
No es necesario comprobar si el id es mayor a cero, siempre debería ser un entero positivo ya que es generado de forma automática.

MatiasOlivera replied:
Igual es solo una observación, no hace falta que lo cambies

MatiasOlivera reviewed (changes requested):
--

MatiasOlivera commented on server/database/migrations/2019_04_01_112937_crear_trigger_before_insert_razones_sociales.php:
```diff
@@ -0,0 +1,33 @@
+     */
+    public function up()
+    {
+        DB::unprepared('CREATE TRIGGER insert_razon_social BEFORE INSERT ON razones_sociales
```
Se podría establecer el valor por defecto `1` en la columna `estado`

MatiasOlivera commented on server/database/migrations/2019_04_11_180910_add_trigger_before_insert_tabla_clientes.php:
```diff
@@ -0,0 +1,34 @@
+     */
+    public function up()
+    {
+        DB::unprepared('CREATE TRIGGER insert_clientes BEFORE INSERT ON clientes
```
Se podría establecer el valor por defecto en las columnas. Usar un trigger es demasiado.

RomeoCirel replied:
pasa que Doctrine no soporta el campo bit para hacer el alter, y como el trigger esta igual por el fecha de carga, pense en ponerlo en el trigger.

RomeoCirel replied:
`$table->boolean('estado')->default(1)->change();`
`Unknown database type bit requested, Doctrine\DBAL\Platforms\MySqlPlatform may not support it.`
voy a volver a buscar si hay solución

MatiasOlivera replied:
Si se modificaba el tipo `bit` a `tinyint`, había que actualizar las consultas en el sistema anterior ya que están utilizando la sentencia `NOT` para calcular el valor binario inverso. Nos conviene dejar los triggers como están para no tener que actualizar ambos sistemas

MatiasOlivera committed:
style: usar camel case al instanciar base controller

MatiasOlivera committed:
style: usar lower case en el constructor

MatiasOlivera committed:
style: usar camel case en las claves de la respuesta

MatiasOlivera committed:
refactor: usar directamente el límite por página que se recibe

Hay que agregar un form request que valide los valores de páginación, ordenamiento, búsqueda, etc.

MatiasOlivera committed:
fix: typo en el campo documento

MatiasOlivera committed:
style: usar camel case en las variables de mensajes

MatiasOlivera committed:
fix: validar que existe el id de localidad

MatiasOlivera committed:
refactor: usar la restricción size en vez de min/max

MatiasOlivera committed:
style: dar formato al código

Sigue el estándar PSR

MatiasOlivera committed:
Merge branch 'develop' into Feature/RegistroCliente
